### DISCUSSION TITLE
M2M fields via API

### DISCUSSION THREAD

**Discussion by H14** [directus/directus#11324] [Archive]
I am a new Directus user, trying to figure out if it could fit our use case.

I have 2 collections with a many to many relation, personen and tags.
![image](https://user-images.githubusercontent.com/892420/151562926-b3d66e99-6d37-4089-9348-829ff67936c3.png)

This seems to work well within the app. 
![image](https://user-images.githubusercontent.com/892420/151563729-e0bf04a2-f01a-423c-b053-58e290ece05e.png)

But I can't seem to get the same (simple) results from the API (the docs on this part are quite condensed and not that hands-on). 
![image](https://user-images.githubusercontent.com/892420/151563788-990a58dc-4c56-42f8-8d55-fe844e5d317a.png)

By default it returns the id's from the **related** collection. If i specifically ask to add everything from tags with the wildcard, I get the fields from the **junction** table, not from the actual related table.
![image](https://user-images.githubusercontent.com/892420/151564095-bdd8af2d-4061-4e72-9b4e-fd8ca30cfd40.png)

The only way I found to get the actual fields of the related table is to reference the junction table in the fields param:
![image](https://user-images.githubusercontent.com/892420/151564781-2e65ddb5-206d-4e02-b0e8-de40c9b634b5.png)

But this brings an extra level of nesting (tags_id) which feels a bit ... weird.

Is there any other way to do this more easily? Or is there a better way to show a list of people and their related tags?

Many thanks!

**Comment by rijkvanzanten** [comment]
Hoi hoi! The above is correct. A many-to-many is no different from a one-to-many followed by a many-to-one. You can have any other fields you might need on the junction collection, as it's just another collection. We can't and won't skip that junction collection by default, as that would cause problems for this use cases.

We could discuss the possibility of adding an additional flag to "skip" collections somehow; curious to hear your thoughts!

Groetjes! 👋🏻

**Reply by lfcc1** [reply]
This would really be appreciated.

**Reply by northern-mindset** [reply]
Just to link issues, another one related to the “skip” feature request: https://github.com/directus/directus/discussions/4112

**Comment by H14** [comment]
Bedankt! Makes sense.

I do believe these features are based on informed decisions, it's just hard to make this stuff out from the docs alone. While the app interface is very friendly and 'no code', it becomes very technical, very fast. Which is fine, but not all field types are self explanatory, and they are just not well documented either. I often have no idea where to start or what steps to take to make something work. Eg: _oh wow, I can even add a tree view, noice_ - followed by many failed attempts to add one because I have no idea what the prerequisites are, what the tree view actually does, what the use case is (and if it fits mine), how it will look like in the api, ... and not a single word about it in the docs. This is just one example, but I'm getting off topic. 

Not bashing, I'm very convinced Directus will be our way forward, but I have coworkers to convince (and they are not the patient type!).

**Reply by jamescammarano** [reply]
Docs are definitely something we need to improve on and we are! it's just a slow rollout right now with one dedicated docs writer and a small team in general. <3
